lib/index.js

```diff
@@ -64,31 +109,42 @@ function render (tree, options) {
   function html (tree) {
     var result = ''
 
-    traverse([].concat(tree), function (node) {
-      if (!node) return
-
-      if (typeof node === 'string' || typeof node === 'number') {
-        result += node
+    tree = Array.isArray(tree) ? tree : [tree]
 
+    traverse(tree, function (node) {
+      // undefined, null, '', [], NaN
+      if (node === undefined ||
+          node === null ||
+          node.length === 0 ||
+          (typeof node === 'number' && isNaN(node))) {
```

Comment:
it would be better `Number.isNaN` or remove https://github.com/posthtml/posthtml-render/blob/7e0b2939a428f379f36ad6b414cdd95bfb2c76c0/lib/index.js#L116 because `isNaN(undefined) => true`